### ISSUE TITLE
Fix FP S6329: vpcSubnets, only raise when public

### ISSUE THREAD

**Issue by pedro-oliveira-sonarsource** [SonarSource/SonarJS#3640]
### **What**

In  #3460 , support for _aws_cdk.aws_rds.DatabaseInstance_ was implemented, where we raise an issue if _publiclyAccessible_ is set to _True_ in the constructor.

On a validation sprint, when analyzing the [code samples](https://github.com/SonarSource/appsec-poc/blob/master/typescript/poc/rules/S6329/src/stacks/databaseinstance_stack.ts) produced during the research for rule specification a false-positive was raising if the _vpcSubnets_ property was not set. When  _vpcSubnets_ is not set, it has a private default value therefore it’s not publicly exposed.

```
// vpc_subnets not defined, "Default: - private subnets"
new rds.DatabaseInstance(this, "public_default_subnet", {
    engine: rds.DatabaseInstanceEngine.POSTGRES,
    vpc: vpc,
    publiclyAccessible: true, // FP - Compliant
    securityGroups: [sg]
})
```

```
// vpc_subnets defined
new rds.DatabaseInstance(this, "public_public_subnet", {
    engine: rds.DatabaseInstanceEngine.POSTGRES,
    vpc: vpc,
    publiclyAccessible: true, // TP - Sensitive
    vpcSubnets: {subnetType: ec2.SubnetType.PUBLIC},  // <-------
    securityGroups: [sg]
})
```

### **How**

When _publiclyAccessible_ is set to true, ensure _vpcSubnets_ is defined to a public type, [the default is private](https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-rds.DatabaseInstance.html#vpcsubnets), [code](https://github.com/SonarSource/SonarJS/blob/2475ffe3caadf4c42c9fb68739010ff228a79dcb/eslint-bridge/src/linting/eslint/rules/aws-ec2-rds-dms-public.ts#L111).
